Source/WebCore/html/canvas/CanvasStyle.cpp

```diff
@@ -47,33 +47,41 @@
 namespace WebCore {
 
 class CanvasStyleColorResolutionDelegate final : public CSSUnresolvedColorResolutionDelegate {
-    WTF_MAKE_FAST_ALLOCATED;
 public:
-    static std::unique_ptr<CanvasStyleColorResolutionDelegate> create(Ref<HTMLCanvasElement> canvasElement)
-    {
-        return WTF::makeUnique<CanvasStyleColorResolutionDelegate>(WTFMove(canvasElement));
-    }
-
     CanvasStyleColorResolutionDelegate(Ref<HTMLCanvasElement> canvasElement)
         : m_canvasElement { WTFMove(canvasElement) }
     {
     }
 
-    Color currentColor() const final
-    {
-        if (!m_canvasElement->isConnected() || !m_canvasElement->inlineStyle())
-            return Color::black;
-
-        auto colorString = m_canvasElement->inlineStyle()->getPropertyValue(CSSPropertyColor);
-        auto color = CSSPropertyParserHelpers::parseColorRaw(WTFMove(colorString), m_canvasElement->cssParserContext(), { });
-        if (!color.isValid())
-            return Color::black;
-        return color;
-    }
+    Color currentColor() const final;
 
     Ref<HTMLCanvasElement> m_canvasElement;
 };
 
+using LazySlowPathColorParsingParameters = std::tuple<
+    CSSPropertyParserHelpers::CSSColorParsingOptions,
+    CSSUnresolvedColorResolutionContext,
+    std::optional<CanvasStyleColorResolutionDelegate>
+>;
+
+Color CanvasStyleColorResolutionDelegate::currentColor() const
+{
+    if (!m_canvasElement->isConnected() || !m_canvasElement->inlineStyle())
+        return Color::black;
+
+    auto colorString = m_canvasElement->inlineStyle()->getPropertyValue(CSSPropertyColor);
+    auto color = CSSPropertyParserHelpers::parseColorRaw(WTFMove(colorString), m_canvasElement->cssParserContext(), [] {
+        return LazySlowPathColorParsingParameters {
+            { },
+            { },
+            std::nullopt
+        };
```

Comment:
Why not horizontal for this?

Reply:
I mean formatting-wise.

Reply:
My thinking at the time was to match the two calls below, but that doesn't hold a lot of water on reflection. Will make it one line.